vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1,1017 +1,1023 @@
+
+      grid._updateItem = tryCatchWrapper(function(row, item) {
+        GridElement.prototype._updateItem.call(grid, row, item);
+
+        // There might be inactive component renderers on hidden rows that still refer to the
+        // same component instance as one of the renderers on a visible row. Making the
+        // inactive/hidden renderer attach the component might steal it from a visible/active one.
+        if (!row.hidden) {
+          // make sure that component renderers are updated
+          Array.from(row.children).forEach(cell => {
+            if(cell._instance && cell._instance.children) {
+              Array.from(cell._instance.children).forEach(content => {
+                if(content._attachRenderedComponentIfAble) {
+                  content._attachRenderedComponentIfAble();
+                }
+              });
+            }
+          });
+        }
+      })
 
-    const itemsUpdated = function(items) {
-      if (!items || !Array.isArray(items)) {
-        throw 'Attempted to call itemsUpdated with an invalid value: ' + JSON.stringify(items);
-      }
-      let detailsOpenedItems = Array.from(grid.detailsOpenedItems);
-      let updatedSelectedItem = false;
-      for (let i = 0; i < items.length; ++i) {
-        const item = items[i];
-        if(!item) {
-          continue;
+      grid._expandedInstanceChangedCallback = tryCatchWrapper(function(inst, value) {
+        // method available only for the TreeGrid server-side component
+        if (inst.item == undefined || grid.$server.updateExpandedState == undefined) {
+          return;
         }
-        if (item.detailsOpened) {
-          if(grid._getItemIndexInArray(item, detailsOpenedItems) < 0) {
-            detailsOpenedItems.push(item);
+        let parentKey = grid.getItemId(inst.item);
+        grid.$server.updateExpandedState(parentKey, value);
+        if (value) {
+          this.expandItem(inst.item);
+        } else {
+          delete cache[parentKey];
+          let parentCache = grid.$connector.getCacheByKey(parentKey);
+          if (parentCache && parentCache.itemkeyCaches && parentCache.itemkeyCaches[parentKey]) {
+            delete parentCache.itemkeyCaches[parentKey];
+          }
+          if (parentCache && parentCache.itemkeyCaches) {
+            Object.keys(parentCache.itemCaches)
+                .filter(idx => parentCache.items[idx].key === parentKey)
+                .forEach(idx => delete parentCache.itemCaches[idx]);
           }
-        } else if(grid._getItemIndexInArray(item, detailsOpenedItems) >= 0) {
-          detailsOpenedItems.splice(grid._getItemIndexInArray(item, detailsOpenedItems), 1)
+          delete lastRequestedRanges[parentKey];
+          this.collapseItem(inst.item);
         }
-        if (selectedKeys[item.key]) {
-          selectedKeys[item.key] = item;
-          item.selected = true;
-          updatedSelectedItem = true;
+      })
+
+      const itemsUpdated = tryCatchWrapper(function(items) {
```

Comment:
This seems to only be called from within other methods that are already using `tryCatchWrapper` so it could be removed here.

Reply:
Updated.